policyd tests fails intermittently

```
2019-11-13 13:03:41 [INFO] ## Running Test zaza.openstack.charm_tests.policyd.tests.NeutronApiTests ##
2019-11-13 13:03:46 [INFO] Using keystone API V3 (or later) for overcloud auth
2019-11-13 13:03:54 [INFO] Using keystone API V3 (or later) for overcloud auth
2019-11-13 13:03:57 [INFO] test_001_policyd_good_yaml (zaza.openstack.charm_tests.policyd.tests.NeutronApiTests)
2019-11-13 13:03:57 [INFO] Test that the policyd with a good zipped yaml file.
2019-11-13 13:03:57 [INFO]  ...
2019-11-13 13:03:57 [INFO] Attaching good zip file as a resource.
2019-11-13 13:03:58 [INFO] Setting config to {'use-policyd-override': 'True'}
2019-11-13 13:03:58 [INFO] Now checking for file contents: /etc/neutron/policy.d/file1.yaml
2019-11-13 13:04:37 [INFO] Checking for workload status line starts with PO:
2019-11-13 13:04:38 [INFO] Disabling policy override by setting config to false
2019-11-13 13:04:38 [INFO] Setting config to {'use-policyd-override': 'False'}
2019-11-13 13:06:10 [INFO] Checking that /etc/neutron/policy.d/file1.yaml has been removed
2019-11-13 13:06:11 [INFO] OK
2019-11-13 13:06:11 [INFO] ok
2019-11-13 13:06:11 [INFO] test_002_policyd_bad_yaml (zaza.openstack.charm_tests.policyd.tests.NeutronApiTests)
2019-11-13 13:06:11 [INFO] Test bad yaml file in the zip file is handled.
2019-11-13 13:06:11 [INFO]  ...
2019-11-13 13:06:11 [INFO] Attaching bad zip file as a resource
2019-11-13 13:06:11 [INFO] Setting config to {'use-policyd-override': 'True'}
2019-11-13 13:06:12 [INFO] Checking for workload status line starts with PO (broken):
2019-11-13 13:07:03 [INFO] Now checking that file /etc/neutron/policy.d/file2.yaml is not present.
2019-11-13 13:07:04 [INFO] Setting config to {'use-policyd-override': 'False'}
2019-11-13 13:07:55 [INFO] OK
2019-11-13 13:07:55 [INFO] ok
2019-11-13 13:07:55 [INFO] test_003_test_overide_is_observed (zaza.openstack.charm_tests.policyd.tests.NeutronApiTests)
2019-11-13 13:07:55 [INFO] Test that the override is observed by the underlying service.
2019-11-13 13:07:55 [INFO]  ...
2019-11-13 13:07:55 [INFO] Doing policyd override for neutron
2019-11-13 13:07:56 [INFO] First verify that operation works prior to override
2019-11-13 13:07:56 [INFO] Authentication for demo on keystone IP 10.5.0.54
2019-11-13 13:07:56 [INFO] keystone IP 10.5.0.54
2019-11-13 13:07:57 [INFO] Doing policyd override with: {'get_network': '!'}
2019-11-13 13:07:57 [INFO] Setting config to {'use-policyd-override': 'True'}
2019-11-13 13:08:52 [INFO] Now verify that operation doesn't work with override
2019-11-13 13:08:52 [INFO] Authentication for demo on keystone IP 10.5.0.54
2019-11-13 13:08:52 [INFO] keystone IP 10.5.0.54
2019-11-13 13:08:53 [INFO] Setting config to {'use-policyd-override': 'False'}
2019-11-13 13:09:36 [INFO] Finally verify that operation works after removing the override.
2019-11-13 13:09:36 [INFO] Authentication for demo on keystone IP 10.5.0.54
2019-11-13 13:09:36 [INFO] keystone IP 10.5.0.54
2019-11-13 13:09:37 [INFO] ERROR
2019-11-13 13:09:37 [INFO] ======================================================================
2019-11-13 13:09:37 [INFO] ERROR: test_003_test_overide_is_observed (zaza.openstack.charm_tests.policyd.tests.NeutronApiTests)
2019-11-13 13:09:37 [INFO] Test that the override is observed by the underlying service.
2019-11-13 13:09:37 [INFO] ----------------------------------------------------------------------
2019-11-13 13:09:37 [INFO] Traceback (most recent call last):
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/urllib3/connectionpool.py", line 672, in urlopen
2019-11-13 13:09:37 [INFO]     chunked=chunked,
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/urllib3/connectionpool.py", line 421, in _make_request
2019-11-13 13:09:37 [INFO]     six.raise_from(e, None)
2019-11-13 13:09:37 [INFO]   File "<string>", line 3, in raise_from
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/urllib3/connectionpool.py", line 416, in _make_request
2019-11-13 13:09:37 [INFO]     httplib_response = conn.getresponse()
2019-11-13 13:09:37 [INFO]   File "/usr/lib/python3.6/http/client.py", line 1346, in getresponse
2019-11-13 13:09:37 [INFO]     response.begin()
2019-11-13 13:09:37 [INFO]   File "/usr/lib/python3.6/http/client.py", line 307, in begin
2019-11-13 13:09:37 [INFO]     version, status, reason = self._read_status()
2019-11-13 13:09:37 [INFO]   File "/usr/lib/python3.6/http/client.py", line 276, in _read_status
2019-11-13 13:09:37 [INFO]     raise RemoteDisconnected("Remote end closed connection without"
2019-11-13 13:09:37 [INFO] http.client.RemoteDisconnected: Remote end closed connection without response
2019-11-13 13:09:37 [INFO] During handling of the above exception, another exception occurred:
2019-11-13 13:09:37 [INFO] Traceback (most recent call last):
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/requests/adapters.py", line 449, in send
2019-11-13 13:09:37 [INFO]     timeout=timeout
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/urllib3/connectionpool.py", line 720, in urlopen
2019-11-13 13:09:37 [INFO]     method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/urllib3/util/retry.py", line 400, in increment
2019-11-13 13:09:37 [INFO]     raise six.reraise(type(error), error, _stacktrace)
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/urllib3/packages/six.py", line 734, in reraise
2019-11-13 13:09:37 [INFO]     raise value.with_traceback(tb)
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/urllib3/connectionpool.py", line 672, in urlopen
2019-11-13 13:09:37 [INFO]     chunked=chunked,
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/urllib3/connectionpool.py", line 421, in _make_request
2019-11-13 13:09:37 [INFO]     six.raise_from(e, None)
2019-11-13 13:09:37 [INFO]   File "<string>", line 3, in raise_from
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/urllib3/connectionpool.py", line 416, in _make_request
2019-11-13 13:09:37 [INFO]     httplib_response = conn.getresponse()
2019-11-13 13:09:37 [INFO]   File "/usr/lib/python3.6/http/client.py", line 1346, in getresponse
2019-11-13 13:09:37 [INFO]     response.begin()
2019-11-13 13:09:37 [INFO]   File "/usr/lib/python3.6/http/client.py", line 307, in begin
2019-11-13 13:09:37 [INFO]     version, status, reason = self._read_status()
2019-11-13 13:09:37 [INFO]   File "/usr/lib/python3.6/http/client.py", line 276, in _read_status
2019-11-13 13:09:37 [INFO]     raise RemoteDisconnected("Remote end closed connection without"
2019-11-13 13:09:37 [INFO] urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))
2019-11-13 13:09:37 [INFO] During handling of the above exception, another exception occurred:
2019-11-13 13:09:37 [INFO] Traceback (most recent call last):
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/keystoneauth1/session.py", line 1004, in _send_request
2019-11-13 13:09:37 [INFO]     resp = self.session.request(method, url, **kwargs)
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/requests/sessions.py", line 533, in request
2019-11-13 13:09:37 [INFO]     resp = self.send(prep, **send_kwargs)
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/requests/sessions.py", line 646, in send
2019-11-13 13:09:37 [INFO]     r = adapter.send(request, **kwargs)
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/requests/adapters.py", line 498, in send
2019-11-13 13:09:37 [INFO]     raise ConnectionError(err, request=request)
2019-11-13 13:09:37 [INFO] requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))
2019-11-13 13:09:37 [INFO] During handling of the above exception, another exception occurred:
2019-11-13 13:09:37 [INFO] Traceback (most recent call last):
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/zaza/openstack/charm_tests/policyd/tests.py", line 494, in get_client_and_attempt_operation
2019-11-13 13:09:37 [INFO]     networks = neutron_client.list_networks()
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/neutronclient/v2_0/client.py", line 818, in list_networks
2019-11-13 13:09:37 [INFO]     **_params)
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/neutronclient/v2_0/client.py", line 369, in list
2019-11-13 13:09:37 [INFO]     for r in self._pagination(collection, path, **params):
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/neutronclient/v2_0/client.py", line 384, in _pagination
2019-11-13 13:09:37 [INFO]     res = self.get(path, params=params)
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/neutronclient/v2_0/client.py", line 354, in get
2019-11-13 13:09:37 [INFO]     headers=headers, params=params)
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/neutronclient/v2_0/client.py", line 331, in retry_request
2019-11-13 13:09:37 [INFO]     headers=headers, params=params)
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/neutronclient/v2_0/client.py", line 282, in do_request
2019-11-13 13:09:37 [INFO]     headers=headers)
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/neutronclient/client.py", line 340, in do_request
2019-11-13 13:09:37 [INFO]     return self.request(url, method, **kwargs)
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/neutronclient/client.py", line 328, in request
2019-11-13 13:09:37 [INFO]     resp = super(SessionClient, self).request(*args, **kwargs)
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/keystoneauth1/adapter.py", line 248, in request
2019-11-13 13:09:37 [INFO]     return self.session.request(url, method, **kwargs)
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/keystoneauth1/session.py", line 913, in request
2019-11-13 13:09:37 [INFO]     resp = send(**kwargs)
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/keystoneauth1/session.py", line 1020, in _send_request
2019-11-13 13:09:37 [INFO]     raise exceptions.ConnectFailure(msg)
2019-11-13 13:09:37 [INFO] keystoneauth1.exceptions.connection.ConnectFailure: Unable to establish connection to http://10.5.0.61:9696/v2.0/networks: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))
2019-11-13 13:09:37 [INFO] During handling of the above exception, another exception occurred:
2019-11-13 13:09:37 [INFO] Traceback (most recent call last):
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/zaza/openstack/charm_tests/policyd/tests.py", line 428, in test_003_test_overide_is_observed
2019-11-13 13:09:37 [INFO]     self.get_client_and_attempt_operation(self.keystone_ips[0])
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/zaza/openstack/charm_tests/policyd/tests.py", line 499, in get_client_and_attempt_operation
2019-11-13 13:09:37 [INFO]     raise PolicydOperationFailedException()
2019-11-13 13:09:37 [INFO] zaza.openstack.charm_tests.policyd.tests.PolicydOperationFailedException
2019-11-13 13:09:37 [INFO] During handling of the above exception, another exception occurred:
2019-11-13 13:09:37 [INFO] Traceback (most recent call last):
2019-11-13 13:09:37 [INFO]   File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/zaza/openstack/charm_tests/policyd/tests.py", line 433, in test_003_test_overide_is_observed
2019-11-13 13:09:37 [INFO]     .format(str(e)))
2019-11-13 13:09:37 [INFO] zaza.openstack.utilities.exceptions.PolicydError: Service action failed and should have passed after removing policy override: ""
2019-11-13 13:09:37 [INFO] ----------------------------------------------------------------------
2019-11-13 13:09:37 [INFO] Ran 3 tests in 355.493s
2019-11-13 13:09:37 [INFO] FAILED
2019-11-13 13:09:37 [INFO]  (errors=1)
Traceback (most recent call last):
  File "/home/ubuntu/src/neutron-api/.tox/func/bin/functest-run-suite", line 8, in <module>
    sys.exit(main())
  File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/zaza/charm_lifecycle/func_test_runner.py", line 162, in main
    bundle=args.bundle)
  File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/zaza/charm_lifecycle/func_test_runner.py", line 107, in func_test_runner
    run_env_deployment(env_deployment, keep_model=preserve_model)
  File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/zaza/charm_lifecycle/func_test_runner.py", line 63, in run_env_deployment
    test_steps.get(deployment.model_alias, []))
  File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/zaza/charm_lifecycle/test.py", line 68, in test
    run_test_list(tests)
  File "/home/ubuntu/src/neutron-api/.tox/func/lib/python3.6/site-packages/zaza/charm_lifecycle/test.py", line 62, in run_test_list
    assert test_result.wasSuccessful(), "Test run failed"
AssertionError: Test run failed
ERROR: InvocationError: '/home/ubuntu/src/neutron-api/.tox/func/bin/functest-run-suite --keep-model'
____________________________________________________ summary ____________________________________________________
ERROR:   func: commands failed
```

--- a/zaza/openstack/charm_tests/policyd/tests.py
+++ b/zaza/openstack/charm_tests/policyd/tests.py
@@ -86,6 +86,7 @@ class PolicydTest(object):
         config = {"use-policyd-override": s}
         logging.info("Setting config to {}".format(config))
         zaza_model.set_application_config(self.application_name, config)
+        zaza_model.wait_for_agent_status()
 
     def _make_zip_file_from(self, name, files):
         """Make a zip file from a dictionary of filename: string.

